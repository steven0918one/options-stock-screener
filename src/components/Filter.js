import React from 'react';
import { useDispatch } from 'react-redux';
import changeFilter from '../actions/filterActions';

function Filter() {
  const filters = ['Reset filters', 'NYSE', 'NASDAQ', 'Other'];
  const dispatch = useDispatch();

  const filterByCategory = e => {
    const { value } = e.target;
    dispatch(changeFilter(value));
  };

  return (
    <div className="Filter">
      <select name="filters" onChange={filterByCategory} className="filter">
        <option default value>
          {' '}
          Filter by Exchange
          {' '}
        </option>
        {filters.map(filter => (
          <option value={filter} key={filter}>
            {filter}
          </option>
        ))}
      </select>
    </div>
  );
}

export default Filter;
