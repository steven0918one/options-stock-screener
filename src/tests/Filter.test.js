import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Filter from '../components/Filter';
import Wrapper from './Wrapper';

Enzyme.configure({ adapter: new Adapter() });

test('Filter should match snapshot', () => {
  const component = renderer.create(
    <Wrapper>
      <Filter />
    </Wrapper>,
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

describe('Filter', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(
      <Wrapper>
        <Filter />
      </Wrapper>,
    );
  });
  it('is receiving and running JS', () => {
    expect(2 + 3).toEqual(5);
  });

  it('renders', () => {
    expect(wrapper).not.toBeNull();
  });

  it('The default value for filter is Filter by Exchange', () => {
    expect(wrapper.find('option').first().text()).toEqual(
      ' Filter by Exchange ',
    );
  });

  it('has an option to reset filters', () => {
    expect(wrapper.find({ value: 'Reset filters' }).text()).not.toBeNull();
  });

  it('has a filter for NASDAQ exchange', () => {
    expect(wrapper.find({ value: 'NASDAQ' }).text()).toEqual('NASDAQ');
  });

  it('has a filter for NYSE exchange', () => {
    expect(wrapper.find({ value: 'NYSE' }).text()).toEqual('NYSE');
  });

  it('has a filter for Other exchange', () => {
    expect(wrapper.find({ value: 'Other' }).text()).toEqual('Other');
  });
});
