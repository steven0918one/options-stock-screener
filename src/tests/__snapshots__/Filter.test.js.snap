// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter is rendered 1`] = `
<div
  className="Filter"
>
  <select
    className="filter"
    name="filters"
    onChange={[Function]}
  >
    <option
      default={true}
      value={true}
    >
       
      Filter by Exchange
       
    </option>
    <option
      value="Reset filters"
    >
      Reset filters
    </option>
    <option
      value="NYSE"
    >
      NYSE
    </option>
    <option
      value="NASDAQ"
    >
      NASDAQ
    </option>
    <option
      value="Other"
    >
      Other
    </option>
  </select>
</div>
`;

exports[`Filter should match snapshot 1`] = `
<div
  className="Filter"
>
  <select
    className="filter"
    name="filters"
    onChange={[Function]}
  >
    <option
      default={true}
      value={true}
    >
       
      Filter by Exchange
       
    </option>
    <option
      value="Reset filters"
    >
      Reset filters
    </option>
    <option
      value="NYSE"
    >
      NYSE
    </option>
    <option
      value="NASDAQ"
    >
      NASDAQ
    </option>
    <option
      value="Other"
    >
      Other
    </option>
  </select>
</div>
`;
